src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,280 @@
+package duke;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.tasks.DeadlineTask;
+import duke.tasks.EventTask;
+import duke.tasks.Task;
+import duke.tasks.ToDoTask;
+
+
+
+/**
+ * TaskList class contains a list in which stores all the tasks that the user has inputted and not deltted
+ */
+public class TaskList {
+
+    /** The list of tasks */
+    private List<Task> inputList = new ArrayList<>();
+    /**
+     * TaskList constructor to intialize a new empty TaskList
+     */
+    public TaskList() {
+        this.inputList = new ArrayList<>();
+    }
+
+    /**
+     * TaskList constructor to intialize a new TaskList with a given ArrayList of tasks
+     * the tasks are all stored as string format which is processed before being
+     * added to the current task list
+     */
+    public TaskList(ArrayList<String> tasks) {
+
+        for (String taskStr: tasks) {
+            String[] arr = taskStr.split("\\|");
+
+            for (int i = 0; i < arr.length; i++) {
+                arr[i] = arr[i].trim();
+            }
+
+            if (arr[0].equals("D")) {
+                String[] deadline = arr[3].split("-");
+                String reformattedDeadline = deadline[2] + "-" + deadline[1] + "-" + deadline[0];
+
+                DeadlineTask deadlineTask = new DeadlineTask(arr[2], reformattedDeadline);
+
+                if (arr[1].equals("1")) {
+                    deadlineTask.markAsDone();
+                }
+
+                inputList.add(deadlineTask);
+
+            } else if (arr[0].equals("E")) {
+                String[] timing = arr[3].split("-");
+                String reformattedTiming = timing[2] + "-" + timing[1] + "-" + timing[0];
+
+                EventTask eventTask = new EventTask(arr[2], reformattedTiming);
+
+                if (arr[1].equals("1")) {
+                    eventTask.markAsDone();
+                }
+
+                inputList.add(eventTask);
+
+            } else {
+                ToDoTask toDoTask = new ToDoTask(arr[2]);
+
+                if (arr[1].equals("1")) {
+                    toDoTask.markAsDone();
+                }
+
+                inputList.add(toDoTask);
+            }
+        }
+    }
+
+    /**
+     * Returns The list of tasks.
+     *
+     * @return The task list
+     */
+    public List<Task> getList() {
+        return this.inputList;
+    }
+
+    /**
+     * Adds a task to the task list
+     *
+     * @param task A task
+     */
+    public void add(Task task) {
+        this.inputList.add(task);
+    }
+
+    /**
+     * Returns The task inside the task list at the index specified
+     *
+     * @param index index of task inside the tasklist
+     * @return A task
+     */
+    public Task get(int index) {

```

Comment:
Perhaps it is better to have a longer method name here e.g getTask ?

Reply:
Removed the method as it is no longer in use